### PR TITLE
Fix: skip syncing blocked onedrive users

### PR DESCRIPTION
**Pull request by R0gue-one**
## Description

- Skips users whose OneDrive is not provisioned, archived/blocked, or fails an availability check without aborting the remaining sync.
- Classifies failures as:
  - OneDrive not provisioned (404)
  - OneDrive archived or blocked (423)
  - Unknown or potentially transient errors

- Sends one warning notification for each failure category.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

## Summary by CodeRabbit

* **New Features**
  * Added clearer OneDrive synchronization status handling for unavailable, unprovisioned, locked, and error-affected drives.
  * Added grouped notifications identifying users whose OneDrive data could not be synchronized, with concise summaries for larger lists.

* **Bug Fixes**
  * Improved warnings for permission issues, locked drives, missing provisioning, and temporary failures.
  * Added fallback notifications when no OneDrive users can be synchronized.
  * Improved notification delivery so relevant synchronization issues are surfaced without limiting alerts to a single recipient.

### CODE REVIEW

File: backend/python/app/connectors/sources/microsoft/onedrive/connector.py [hunk 1]
```diff
@@ -1285,13 +1294,27 @@ async def _process_users_in_batches(self, users: List[AppUser]) -> None:
...
-            # Further filter to only users with OneDrive provisioned
             users_to_sync = []
+            failed_users: List[Tuple[str, OneDriveUserStatus]] = []
             for user in active_users:
-                if await self._user_has_onedrive(user.source_user_id):
+                try:
+                    onedrive_status = await self._user_has_onedrive(user.source_user_id)
+                except Exception as e:
+                    self.logger.error(f"❌ Error checking OneDrive for user {user.email}, skipping: {e}", exc_info=True)
+                    failed_users.append((user.email, OneDriveUserStatus.UNKNOWN_ERROR))
```
[the remark this answers is not in the document]
Comment (R0gue-one) [reply backend/python/app/connectors/sources/microsoft/onedrive/connector.py:1324]: a 404 can mean both either we don't have Files.Read.All permission or onedrive is actually not provisioned. But if Files.Read.All is genuinely not provided then all users will fail no 0 users will be synced. 

thats why we return the notify https://github.com/pipeshub-ai/pipeshub-ai/blob/1319ccef9359e2c1b7a84b67ce2b72161962938b/backend/python/app/connectors/sources/microsoft/onedrive/connector.py#L1321-L1335 That either genuinly no users have one drive or Files.Read.All permission is missing.

**Review by abhishek-pipeshub — APPROVED** [review]

**Review by abhishekgml1991 — APPROVED** [review]
